registration-processor/core-processor/registration-processor-abis-handler-stage/src/main/java/io/mosip/registration/processor/abis/handler/stage/AbisHandlerStage.java

```diff
@@ -599,6 +618,80 @@ public String getFilename() {
 
 		return response.getDataShare().getUrl();
 	}
+	private void validateBiometricRecord(BiometricRecord biometricRecord, List<String> modalities, Map<String, String> metaInfoMap) throws DataShareException, JsonParseException, JsonMappingException, IOException {
+		
+		if(modalities==null || modalities.isEmpty()) {
+			throw new DataShareException("Data Share Policy Modalities were Empty");
+
+		}
+		if(biometricRecord == null  || biometricRecord.getSegments() == null
+				|| biometricRecord.getSegments().isEmpty()) {
+			throw new DataShareException("No Biometrics Found with Data Share Policy");
+
+		}	
+		
+		String exceptionBometricsMap = metaInfoMap.get("exceptionBiometrics");
+		
+
+		 ObjectMapper mapper = new ObjectMapper();
+		Map<String, Map<String,Object>> metaInfoExceptionBiometrics = mapper.readValue(exceptionBometricsMap, new TypeReference<Map<String, Map<String,Object>>>() {
+		        });
+		
+		
+		Set<String> exceptionList =  metaInfoExceptionBiometrics.get("applicant").keySet();
+		
+
+		boolean isBioFound = false;
+		for(String modality : modalities) {
+			
+			if(biometricModalitySegmentsMap.containsKey(modality)) {

```

Comment:
check with a inverted condition and throw exception if matches to avoid the below code nesting

Reply:
Incorporated review comments